src/RA_Dlg_MemBookmark.cpp

```diff
@@ -750,42 +724,43 @@ void Dlg_MemBookmark::ExportJSON()
         return;
 
     rapidjson::Document doc;
-    auto& allocator{ doc.GetAllocator() };
+    auto& allocator{doc.GetAllocator()};
     doc.SetObject();
 
-    rapidjson::Value bookmarks{ rapidjson::kArrayType };
+    rapidjson::Value bookmarks{rapidjson::kArrayType};
 
-    for (auto bookmark : m_vBookmarks)
+    for (const auto& bookmark : m_vBookmarks)
     {
-        rapidjson::Value item{ rapidjson::kObjectType };
-        std::string str{ ra::Narrow(bookmark->Description()) };
-        rapidjson::Value s{ str.c_str(), allocator };
+        rapidjson::Value item{rapidjson::kObjectType};
+        std::string str{ra::Narrow(bookmark.Description())};
+        rapidjson::Value s{str.c_str(), allocator};
 
         item.AddMember("Description", s, allocator);
-        item.AddMember("Address", bookmark->Address(), allocator);
-        item.AddMember("Type", bookmark->Type(), allocator);
-        item.AddMember("Decimal", bookmark->Decimal(), allocator);
+        item.AddMember("Address", bookmark.Address(), allocator);
+        item.AddMember("Type", bookmark.Type(), allocator);
+        item.AddMember("Decimal", bookmark.Decimal(), allocator);
         bookmarks.PushBack(item, allocator);
     }
     doc.AddMember("Bookmarks", bookmarks, allocator);
 
-    swprintf_s(buf, L"%s", ofn.lpstrFile);
+    Ensures(swprintf_s(buf, BUF_SIZE, L"%s", ofn.lpstrFile) >= 0);
     _WriteBufferToFile(buf, doc);
 }
 
-void Dlg_MemBookmark::ImportFromFile(std::wstring sFilename) 
+void Dlg_MemBookmark::ImportFromFile(std::wstring&& sFilename)
 {
-    std::ifstream ifile{ sFilename };
+    const auto file_name{std::move(sFilename)};
+    std::ifstream ifile{file_name};
     if (!ifile.is_open())
     {
-        char buf[2048U]{};
-        strerror_s(buf, errno);
-        RA_LOG("%s: %s", ra::Narrow(sFilename).c_str(), buf);
+        std::array<char, 2048> buf{};
+        strerror_s(buf.data(), sizeof(buf), errno);
+        RA_LOG("%s: %s", ra::Narrow(sFilename).c_str(), buf.data());
```

Comment:
I could have used `std::string`, but the internal pointer is dynamically allocated on the free-store.